soc/silabs/silabs_s2/config/sl_device_init_dcdc_config.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This configuration header is used by the HAL driver device_init_dcdc from hal_silabs,
+ * invoked through the soc_early_init hook. DeviceTree options are converted to config macros
+ * expected by the HAL driver.
+ */
```

Comment:
> shouldn;t this code be part of modules/hal_silabs? All glue code should reside in modules/

I'll let @asmellby comment on this specific SoC change, but in general we have a separate task of moving glue code out from the hal_silabs tree into `modules/hal_silabs` in the main tree (note: this directory doesn't even exist yet in the main tree). This will also include moving all Kconfig and CMake files from the `hal_silabs` tree to `modules/hal_silabs` in the main tree (i.e. we'll be adding `kconfig-ext: True` and `cmake-ext: True` to our `zephyr/module.yml`). Anyway that kind of refactoring is unrelated and out of scope for this PR.

Reply:
Agree that modules/ may be the better place, and that should be handled by the later refactor. I added these files in soc/silabs/ since there already was an existing config file like this there (sl_device_init_hfxo_config.h).